### PR TITLE
feat: add draft message data to chat

### PR DESCRIPTION
**Pull request by nkvch**
@udislau

https://github.com/10clouds/aiconsole/assets/58534398/f0b24d0a-770d-4e66-a74f-645c95581a22

### CODE REVIEW

**Review by udislau — CHANGES_REQUESTED** [review]
Draft during execution disappears

https://github.com/10clouds/aiconsole/assets/30052145/3445d9de-7f9f-4c79-981b-37318dde8e2f

**Comment by nkvch** [comment]
I see. I've figured out the origin of the problem. Will fix or workaround it tomorrow morning

**Comment by udislau** [comment]
> I see. I've figured out the origin of the problem. Will fix or workaround it tomorrow morning

Plz don't do any workarounds, you have time to implement it properly till 6pm 🙂

**Comment by nkvch** [comment]
Okay, the issue fixed. Though I am not too sure about the purpose of chat locking mechanism, so I hope I didn't break anything :sweat_smile:. At least from what I see it works properly.
The other idea was to create a separate API service for draft messages, but I saw it as overkill

File: backend/aiconsole/core/chat/locking.py [hunk 1]
```diff
@@ -67,7 +68,11 @@ async def _read_chat_outside_of_lock(chat_id: str):
     if chat_id not in chats:
         return await load_chat_history(chat_id)
 
-    return chats[chat_id]
+    locked_chat = chats[chat_id]
+    current_draft_message = await load_current_draft_message(chat_id)
```
Comment (nkvch) [inline backend/aiconsole/core/chat/locking.py:72]: @udislau is this ok?

**Comment by cielecki** [comment]
draft_message should be part of chat_options

**Pull request closed without merging**